src/cluster/kv/etcd/store.go

```diff
@@ -616,7 +617,29 @@ func (c *client) writeCacheToFile() error {
 	return nil
 }
 
-func (c *client) initCache() error {
+func (c *client) createCacheDir(fm os.FileMode) error {
+	path := path.Dir(c.opts.CacheFileFn()(c.opts.Prefix()))
+	if err := os.MkdirAll(path, fm); err != nil {
+		c.m.diskWriteError.Inc(1)
+		c.logger.Warn("error creating cache directory",
+			zap.String("path", path),
+			zap.Error(err),
+		)
+	}
```

Comment:
I think you want to `return err` here, otherwise it will flow through to the create successful path no?

Reply:
Absolutely! Forgot that to put back in after refactoring, fixed now.